test/controllers/FileUploadControllerSpec.scala

```diff
@@ -201,6 +199,20 @@ class FileUploadControllerSpec extends UnitSpec with I18nHelper with RasTestHelp
         val result = await(TestFileUploadController.uploadError().apply(uploadRequest))
         redirectLocation(result).get should include("/upload-a-file")
       }
+
+      "no file is selected to upload" in {
+        val result = await(TestFileUploadController.post().apply(fakeRequest.withHeaders(("Content-Length","0"))))
+        status(result) shouldBe BAD_REQUEST
+        doc(result).getElementById("header").text shouldBe Messages("file.upload.page.header")
+
+      }
+
+      "a non csv file is selected to upload" in {
+        val file = FilePart("file","test.xml",Some("text.xml"), TemporaryFile("uploaded"))
+        val result = await(mockTestFileUploadController(file).post().apply(fakeRequest.withHeaders(("Content-Length","100"))))
+        status(result) shouldBe BAD_REQUEST
+        doc(result).getElementById("header").text shouldBe Messages("file.upload.page.header")
+      }
```

Comment:
I think you're missing an 'EntityTooLarge' test case here?

Reply:
Good call, added now